source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -155,42 +474,50 @@ namespace aspect
+              relative_support_point_location[d] -= unit_offset;
             for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
               {
-                double interpolated_value = c[property_index][0] +
-                                            c[property_index][1] * relative_support_point_location[0] +
-                                            c[property_index][2] * relative_support_point_location[1];
-                if (dim == 2)
-                  {
-                    interpolated_value += c[property_index][3] * relative_support_point_location[0] * relative_support_point_location[1] +
-                                          c[property_index][4] * relative_support_point_location[0] * relative_support_point_location[0] +
-                                          c[property_index][5] * relative_support_point_location[1] * relative_support_point_location[1];
-                  }
-                else
+                if (selected_properties[property_index] == true)
                   {
-                    interpolated_value += c[property_index][3] * relative_support_point_location[2] +
-                                          c[property_index][4] * relative_support_point_location[0] * relative_support_point_location[1] +
-                                          c[property_index][5] * relative_support_point_location[0] * relative_support_point_location[2] +
-                                          c[property_index][6] * relative_support_point_location[1] * relative_support_point_location[2] +
-                                          c[property_index][7] * relative_support_point_location[0] * relative_support_point_location[0] +
-                                          c[property_index][8] * relative_support_point_location[1] * relative_support_point_location[1] +
-                                          c[property_index][9] * relative_support_point_location[2] * relative_support_point_location[2];
+                    double interpolated_value = evaluate_interpolation_function(c[property_index], relative_support_point_location);
+                    // Overshoot and undershoot correction of interpolated particle property.
+                    if (use_quadratic_least_squares_limiter[property_index])
+                      {
+                        Assert(interpolated_value >= property_minimums[property_index] - std::max(std::abs(property_minimums[property_index]), std::abs(property_maximums[property_index])) * 10. * std::numeric_limits<double>::epsilon(), ExcInternalError());
+                        Assert(interpolated_value <= property_maximums[property_index] + std::max(std::abs(property_minimums[property_index]), std::abs(property_maximums[property_index])) * 10. * std::numeric_limits<double>::epsilon(), ExcInternalError());
+
+                        interpolated_value = std::min(interpolated_value, property_maximums[property_index]);
```

Comment:
Add a comment here that you do that to avoid values that are just barely outside the interpolation bounds.